### DISCUSSION TITLE
Where to find the pinout?

### DISCUSSION THREAD

**Discussion by ReDetection** [stm32duino/Arduino_Core_STM32#1615] [Q&A] [answered]
Many cores publish their pinouts in a nice and readable image, e.g. https://github.com/MCUdude/MiniCore#pinout. Unfortunately I wasn't able to spot one here – did I search it the wrong way? It is not very convenient to lookup in the datasheet every time and a quick overview would be really helpful. Especially when you use boards from different suppliers – they tend to put similar information on the different sections of the datasheet.

E.g. if I am to google the bluepill pinout, I find at least 3 different pinouts on the first page.

**Comment by fpistm** [comment, marked as answer]
Hi  @ReDetection 
You're right several cores provides this. This core provides near than 600 boards definitions (including generic) compare to other cores which manages few boards. 
Anyway this is a thing I have in mind but not so easy to deploy and maintains. Goal would be to generate it based on variant definition.
At this time you can refers to the varian_*.h definition and/or this site: https://stm32-base.org/boards/
Ex for BluePill: https://stm32-base.org/boards/STM32F103C8T6-Blue-Pill

**Reply by ReDetection** [reply]
Hi @fpistm. Thank you for the answer. Yes I do acknowledge the variety of boards – I am also very grateful you can support that many.

I do have my doubts about autogeneration based on just variant_* files because it is not just pinout, it is also pwm/communications/additional functions of the mcu, alternative pinouts, current limitations, notes, etc. This would probably require some additional configuration file to do it right. But I see a nice side too:
* this file is not required for the variant to be accepted into the repo, it will just improve readability and reduce entry level
* file could be contributed by other person :) I would gladly support that

I wonder how this pinouts could be discovered without graphical solutions but have no idea so far

**Reply by fpistm** [reply]
About functions the peripherals pins give use those information.
But I agree it will not so easy and requires some addition :)
